packages/components/src/tree-select/types.ts

```diff
@@ -1,29 +1,22 @@
-/**
- * External dependencies
- */
-import type { ComponentProps } from 'react';
-
 /**
  * Internal dependencies
  */
-import type SelectControl from '../select-control';
-import type { SelectControlProps } from '../select-control/types';
+import type { SelectControlSingleSelectionProps } from '../select-control/types';
 
 export type Truthy< T > = T extends false | '' | 0 | null | undefined
 	? never
 	: T;
 
-export type SelectOptions = Required<
-	ComponentProps< typeof SelectControl >
->[ 'options' ];
-
 export interface Tree {
 	id: string;
 	name: string;
 	children?: Tree[];
 }
 
-export interface TreeSelectProps extends Omit< SelectControlProps, 'value' > {
+// `TreeSelect` inherits props from `SelectControl`, but only
+// in single selection mode (ie. when the `multiple` prop is not defined).
+export interface TreeSelectProps
+	extends Omit< SelectControlSingleSelectionProps, 'value' | 'multiple' > {
```

Comment:
We can omit the `multiple` prop from `TreeSelect` (since it doesn't seem to be relevant to the component). This simplified the component's prop types and removes a bunch of TS errors, allowing us to remove hacks like the ones in the `QueryControls` component